Neos.Neos/Classes/Service/HtmlAugmenter.php

```diff
@@ -62,7 +62,8 @@ public function addAttributes($html, array $attributes, $fallbackTagName = 'div'
      */
     protected function getHtmlRootElement($html)
     {
-        if (trim($html) === '') {
+        $html = trim($html);
```

Comment:
The getHtmlRootElement method only determines wether a root element exists and what tagname it has. If a root element is detected the replacement happens with preg_replace anyways. 

The test with "   <p>Simple HTML with unique root element in whitespace</p>   " verifies exactly that case.